src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2338,18 +2337,59 @@ MavlinkReceiver::handle_message_hil_sensor(mavlink_message_t *msg)
 void
 MavlinkReceiver::handle_message_hil_gps(mavlink_message_t *msg)
 {
-	// Only use to set home position
-	if (_hil_pos_ref.isInitialized()) {
-		return;
-	}
-
 	mavlink_hil_gps_t hil_gps;
 	mavlink_msg_hil_gps_decode(msg, &hil_gps);
 
-	_hitl_sim_gps_time_usec = hil_gps.time_usec;
-	_hitl_sim_home_lat = hil_gps.lat;
-	_hitl_sim_home_lon = hil_gps.lon;
-	_hitl_sim_home_alt = hil_gps.alt;
+	sensor_gps_s gps{};
```

Comment:
Looks like this is unrelated ?

Reply:
Yes, I had temporarily reverted one patch which breaks the sitl. Will remove that!